app/controllers/user.js

```diff
@@ -479,5 +496,47 @@ module.exports = {
     } catch (error) {
       HANDLER.handleError(error)
     }
+  },
+  // Find User and Create if doesn't exist
+  findOrCreateForOAuth: async(profile, provider) => {
+    let user;
+    if(provider==='google'){
+      user = formatUser.formatUser(profile, provider)
+    }
+    try {
+      const existingUser = await User.findOne({
+        email: user.email
+      })
+      if (existingUser) {
+        const token = await existingUser.generateAuthToken()
+        return  {token, user: existingUser};
+      } else {
+        const newUser = new User(user)
+        const isRegisteredUserExists = await User.findOne({ firstRegister: true })
+        const Org = await Organization.find({}).lean().exec()
+        // for the first user who will be setting up the platform for their community
+        if (!isRegisteredUserExists) {
+          newUser.isAdmin = true
+          newUser.firstRegister = true
+        }
+        if (Org.length > 0) {
+          newUser.orgId = Org[0]._id
+        }
+        const data = await newUser.save()
+        if (!isRegisteredUserExists) {
+          settingHelper.addAdmin(data._id)
+        }
+        const token = await newUser.generateAuthToken()
+
+        // create redis db for activity for the user
+        const activity = new Activity({ userId: data._id })
+        await activity.save()
+        // hide password
+        data.password = undefined
+        return {token: token, user: data}
+      }
+    } catch(e) {
+      throw e;
```

Comment:
please handle the exception in a better way, please check the codebase

Reply:
Yes, you're right! my error handling, in this case, is not effective as I'm new to callback things.
I'll read a few articles and rectify them with best practices.

Reply:
Not done yet!

Reply:
How may we proceed with handling error over here?
Should I use `next(err, false)`?

Reply:
Hey please check the codebase, I have written some error handler, you just need to pass the error to that.

Reply:
@Rupeshiya, the reason why I'm throwing error in user.js because I'm catching the same in [passportOAuth.js](https://github.com/codeuino/social-platform-donut-backend/blob/1ad2ced6fe53fecc500ea8993875c2f0db77bfdd/app/middleware/passportOAuth.js) and from that passing it in the verify function(next()) which handles the error in [OAuthMiddlewares](https://github.com/codeuino/social-platform-donut-backend/blob/1ad2ced6fe53fecc500ea8993875c2f0db77bfdd/app/middleware/OAuthMiddlewares.js).
I see that the errorHandler sends a response to the client directly.

Reply:
not only that, we need to redirect the user to the login page since it's OAuth. ErrorHandler just sends JSON to Client Side.